src/dev_server.rs

```diff
@@ -4,31 +4,73 @@ use clap::Parser;
 use std::{
     fs,
     io::{prelude::*, BufReader},
-    net::{IpAddr, SocketAddr, TcpListener, TcpStream},
+    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener, TcpStream},
     path::Path,
     process,
 };
 
+/// A simple HTTP server that can, optionally, watch a given command.
+///
+/// Get the files at `watch_path` and serve them at a given IP address
+/// (127.0.0.1:8000 by default). An optional command can be run along the
+/// server, relaunched if changes are detected in your project via a [`Watch`].
+///
+/// # Usage
+///
+/// ```no_run
+/// use xtask_wasm::{metadata, DevServer};
+/// use std::process::Command;
+///
+/// let build_path = metadata().workspace_root.join("build");
+/// let mut command = Command::new("cargo");
+/// command.arg("run");
+///
+/// DevServer::new()
+/// .command(command)
+/// .start(build_path)
+/// .expect("cannot run dev server process");
+/// ```
+///
+/// This starts to serve files in the `build` directory, run the `cargo run`
+/// command. If changes are detected in the workspace root (recursively
+/// expect for target directory or hidden files), the `cargo run` command is
+/// killed and re-launched.
 #[non_exhaustive]
 #[derive(Debug, Parser)]
 pub struct DevServer {
+    /// Ip address to bind. Default to `127.0.0.1`
     #[clap(long, default_value = "127.0.0.1")]
     pub ip: IpAddr,
+    /// Port number. Default to `8000`
     #[clap(long, default_value = "8000")]
     pub port: u16,
 
-    #[clap(flatten)]
-    pub watch: Watch,
+    /// Watched command running along the server
     #[clap(skip)]
     pub command: Option<process::Command>,
+    /// Watching process of the server
+    #[clap(flatten)]
+    pub watch: Watch,
 }
 
 impl DevServer {
+    /// Create a new `DevServer`.
+    pub fn new() -> Self {
+        Self {
+            ip: IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+            port: 8_000,
+            command: None,
+            watch: Watch::new(),
+        }
+    }
```

Comment:
It might be useful for the user to create instances of these structs (DevServer, Build, etc...) manually (without clap).

Unfortunately this is out-of-scope with this PR and the proposed solution here isn't great anyway.

My guess is that you made those because you wrote doc tests and you needed to instantiate them somehow. You can use [`parse_from`](https://docs.rs/clap/latest/clap/trait.Parser.html#method.parse_from) for that.

```suggestion
```